From a689a015e67d83bd5af37d0029764ccf66cb5aad Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sat, 17 Aug 2013 13:47:53 -0500
Subject: [PATCH 170/390] rtl8192du: Fix build problem for kernels older than
 2.6.35

There is an unbalanced number of parens for older kernels.

From: jflemingeds through notifications@github.com
Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_mlme.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_mlme.c
+++ b/drivers/staging/rtl8192du/core/rtw_mlme.c
@@ -2286,12 +2286,12 @@ void rtw_dynamic_check_timer_handlder(st
 #endif /*  (LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 35)) */
 
 #if (LINUX_VERSION_CODE <= KERNEL_VERSION(2, 6, 35))
-	if (adapter->pnetdev->br_port)
+	if (adapter->pnetdev->br_port &&
+	    check_fwstate(pmlmepriv, WIFI_STATION_STATE | WIFI_ADHOC_STATE)) {
 #else /*  (LINUX_VERSION_CODE <= KERNEL_VERSION(2, 6, 35)) */
-	if (rcu_dereference(adapter->pnetdev->rx_handler_data)
+	if (rcu_dereference(adapter->pnetdev->rx_handler_data) &&
+	    check_fwstate(pmlmepriv, WIFI_STATION_STATE | WIFI_ADHOC_STATE)) {
 #endif /*  (LINUX_VERSION_CODE <= KERNEL_VERSION(2, 6, 35)) */
-	    && (check_fwstate(pmlmepriv, WIFI_STATION_STATE | WIFI_ADHOC_STATE)
-		== true)) {
 		/*  expire NAT2.5 entry */
 		nat25_db_expire(adapter);
 
